components/builder-events/src/error.rs

```diff
@@ -29,6 +31,10 @@ impl From<EventBuilderError> for Error {
     fn from(err: EventBuilderError) -> Error { Error::CloudEventBuilderError(err) }
 }
 
+impl From<rdkafka::error::KafkaError> for Error {
+    fn from(err: rdkafka::error::KafkaError) -> Error { Error::KafkaError(err) }
+}
+
```

Comment:
:thinking: To keep things general like we've been discussing implies that the error handling should be similarly general.

You could require that all your Event Bus implementors return an `EventBusError`, which could be defined generically in terms of `Box<dyn std::error::Error>`. This way, the implementors can create whatever kinds of event structure they want to, and as long as it implements `std::error::Error` (which it will), things will be fine. Importantly, _you_ won't have to keep adding new error variants here to accommodate new event backend implementations.